mitodl/mitxonline
# Updates "skipped" to "denied" in flexible pricing.

#### Pre-Flight checklist


#### What are the relevant tickets?

#756

#### What's this PR do?

Changes the `skipped` status in Flexible Pricing to `denied`. This includes a migration to change any existing `skipped` records into `denied` ones (and vice versa if it needs to be reverted). Furthermore, this includes a small cosmetic change to format the statuses in the Staff Dashboard in a more human-friendly format (so you see "Approved" or "Pending Manual Approval" instead of "approved" and "pending-manual-approval"). 

#### How should this be manually tested?

Automated tests should pass.

Testing database (looking in either the database directly or via Django Admin):
1. Check existing FlexiblePrice records for a record in `skipped` status. (Add one if necessary.) 
2. Run the migration.
3. The status(es) should now be set to `denied`. In addition, loading the Flexible Pricing request form for the given request should show the denied message as per usual, and no discount should be offered to the user. 

Testing in Staff Dashboard:
1. From the Staff Dashboard sidenav, choose Flexible Pricing.
2. The filter dropdown at the top should list Denied as an option for filtering by Status, and choosing that status should show only Denied requests (including any requests that were Skipped).
3. Clicking the Deny button on an existing request should work as expected.
